openerp/models.py

```diff
@@ -5306,6 +5306,10 @@ def with_env(self, env):
         """ Returns a new version of this recordset attached to the provided
         environment
 
+        .. warning::
+            The new environment may not benefit from the initial environment's
+            data cache. This may introduce performance problems.
```

Comment:
May I suggest a wording change?

> The new environment will not benefit from the current environment's
> data cache, so later data access may incur extra delays while re-fetching
> from the database.

My main point is that it doesn't introduce "problems", it *may* cause extra delays that *may or may not* be a problem (user-facing actions vs background jobs, etc.) The documentation should just state the facts, the developer should infer the consequences according to his particular application.